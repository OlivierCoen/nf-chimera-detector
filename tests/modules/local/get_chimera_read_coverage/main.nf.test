nextflow_process {

    name "Test Process GET_CHIMERA_READ_COVERAGE"
    script "modules/local/get_chimera_read_coverage/main.nf"
    process "GET_CHIMERA_READ_COVERAGE"
    tag "get_chimera_coverage"

    test("Chimera found") {

        when {
            process {
                """
                input[0] = [
                    [ id: "test" ],
                    file( "${projectDir}/tests/test_data/blast_outputs/chimeras/DRR515383.all_hits.txt", checkIfExists: true),
                    file( "${projectDir}/tests/test_data/chimeras/DRR515383_chimeras.csv", checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Empty chimera file - Success if fails") {

        when {
            process {
                """
                input[0] = [
                    [ id: "test" ],
                    file( "${projectDir}/tests/test_data/blast_outputs/chimeras/blastOnTEs.txt", checkIfExists: true),
                    file( "${projectDir}/tests/test_data/chimeras/empty_chimeras.csv", checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert !process.success
        }

    }

}
