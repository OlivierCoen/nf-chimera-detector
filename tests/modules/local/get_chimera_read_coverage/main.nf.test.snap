{
    "Chimera found": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                "log.png:md5,a16ff5f0e13016f27aa82d765769d1fb",
                                "raw.png:md5,d6b32b018142681bbf944d064bfde98c"
                            ],
                            [
                                "log.png:md5,ff380b710deb1febc9d027d6f7aa7e45",
                                "raw.png:md5,206daa183f8a23c7b5f2edad50b59fea"
                            ],
                            [
                                "log.png:md5,1c2492b152911731bb358f538a10abc7",
                                "raw.png:md5,a7cebb2383cff3fc4e6dd047341b0a81"
                            ],
                            [
                                "log.png:md5,fe41163fc3c9d0a4db6ce841c03e458f",
                                "raw.png:md5,147586fef9529a05d5da87b92a239647"
                            ],
                            [
                                "log.png:md5,95571a5e8ed38b0af34c142e305ac021",
                                "raw.png:md5,5792c60ebe41ee88d49a19f09c43aa50"
                            ],
                            [
                                "log.png:md5,fdfd79f09e8d3689e2d6da730c18ed74",
                                "raw.png:md5,cbc61340cb4e7cf7d26b74776f316826"
                            ]
                        ]
                    ]
                ],
                "1": [
                    [
                        "GET_CHIMERA_READ_COVERAGE",
                        "python",
                        "3.14.0"
                    ]
                ],
                "2": [
                    [
                        "GET_CHIMERA_READ_COVERAGE",
                        "pandas",
                        "2.3.3"
                    ]
                ],
                "3": [
                    [
                        "GET_CHIMERA_READ_COVERAGE",
                        "polars",
                        "1.34.0"
                    ]
                ],
                "4": [
                    [
                        "GET_CHIMERA_READ_COVERAGE",
                        "matplotlib",
                        "3.10.7"
                    ]
                ],
                "5": [
                    [
                        "GET_CHIMERA_READ_COVERAGE",
                        "tqdm",
                        "4.67.1"
                    ]
                ],
                "coverage_dir": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                "log.png:md5,a16ff5f0e13016f27aa82d765769d1fb",
                                "raw.png:md5,d6b32b018142681bbf944d064bfde98c"
                            ],
                            [
                                "log.png:md5,ff380b710deb1febc9d027d6f7aa7e45",
                                "raw.png:md5,206daa183f8a23c7b5f2edad50b59fea"
                            ],
                            [
                                "log.png:md5,1c2492b152911731bb358f538a10abc7",
                                "raw.png:md5,a7cebb2383cff3fc4e6dd047341b0a81"
                            ],
                            [
                                "log.png:md5,fe41163fc3c9d0a4db6ce841c03e458f",
                                "raw.png:md5,147586fef9529a05d5da87b92a239647"
                            ],
                            [
                                "log.png:md5,95571a5e8ed38b0af34c142e305ac021",
                                "raw.png:md5,5792c60ebe41ee88d49a19f09c43aa50"
                            ],
                            [
                                "log.png:md5,fdfd79f09e8d3689e2d6da730c18ed74",
                                "raw.png:md5,cbc61340cb4e7cf7d26b74776f316826"
                            ]
                        ]
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-17T15:38:50.915073337"
    }
}